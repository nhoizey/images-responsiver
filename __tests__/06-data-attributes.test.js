'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('data attributes', () => {
	test('both src and data-src, only src used', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test1.png" data-src="test3.png" /></body></html>`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('both data-src and data-srcset, no action', () => {
		const content = `<!DOCTYPE html><html><body>
      <img data-src="test1.png" data-srcset="test1-300.png 300w, test1-600.png 600w" /></body></html>`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('data-src but no src nor data-srcset', () => {
		const content = `<!DOCTYPE html><html><body>
      <img data-src="test.png" width="420"></body></html>`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
