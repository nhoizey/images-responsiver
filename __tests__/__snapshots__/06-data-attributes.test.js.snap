// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data attributes both data-src and data-srcset, no action 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-src=\\"test1.png\\"
      data-srcset=\\"test1-300.png 300w, test1-600.png 600w\\"
    />
  </body>
</html>
"
`;

exports[`data attributes both src and data-src, only src used 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test1.png\\"
      sizes=\\"100vw\\"
      srcset=\\"
        test1-320.png   320w,
        test1-560.png   560w,
        test1-800.png   800w,
        test1-1040.png 1040w,
        test1-1280.png 1280w
      \\"
      src=\\"test1-640.png\\"
      data-src=\\"test3.png\\"
    />
  </body>
</html>
"
`;

exports[`data attributes data-src but no src nor data-srcset 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      data-srcset=\\"test-320.png 320w, test-420.png 420w\\"
      data-src=\\"test-420.png\\"
      width=\\"420\\"
    />
  </body>
</html>
"
`;
